File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int cur = 100001;
        for(int price: prices) {
            if(price<cur) {
                cur = price;
            } else {
                int profit = price-cur;
                maxProfit = Math.max(profit,maxProfit);
            }
        }
        return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int ans = 0;
        if(p.size() <= 1) return 0;
        int cur = p[0];
        for(int i=1;i<p.size();i++) {
            if(p[i] > cur) {
                int prof = p[i]-cur;
                ans = max(ans,prof);
            } else {
                cur = p[i];
            }
        }
        return ans;
    }
};